tests/rados/test_mon_system_operations.py

```diff
@@ -0,0 +1,272 @@
+
+    log.info("---------- starting workflow 1 - Addition of new mons --------")
+    # Setting the mon service as unmanaged by cephadm
+    if not mon_obj.set_mon_service_managed_type(unmanaged=True):
+        log.error("Could not set the mon service to unmanaged")
+        raise Exception("mon service not unmanaged error")
+
+    found = False
+    # Idenitfying test host without any mons to deploy new mons
+    for item in cluster_nodes:
+        if item.hostname not in [mon.hostname for mon in init_mon_nodes]:
+            alt_mon_host = item
+            found = True
+            break
+    if not found:
+        log.error("No alternate mon host was found... fail")
+        raise Exception("No free mon host found error")
+
+    log.debug(
+        f"Selected host: {alt_mon_host.hostname} with "
+        f"IP : {alt_mon_host.ip_address} to deploy new mon daemon"
+    )
+
+    # Adding the new mon to the cluster
+    if not mon_obj.add_mon_service(host=alt_mon_host):
+        log.error(f"Could not add mon service on host {alt_mon_host.hostname} ")
+        raise Exception("mon service not added error")
+
+    # Setting the mon service as managed by cephadm
+    if not mon_obj.set_mon_service_managed_type(unmanaged=False):
+        log.error("Could not set the mon service to managed")
+        raise Exception("mon service not managed error")
+
+    # Checking if the new mon added is part of Mon Quorum
+    quorum = mon_obj.get_mon_quorum_hosts()
+    if alt_mon_host.hostname not in quorum:
+        log.error(
+            f"selected host : {alt_mon_host.hostname} does not have mon as part of Quorum post addition"
+        )
+        raise Exception("Mon not in quorum error")
+
+    log.info("Completed workflow 1. Addition of new mon daemon")
+
+    log.info("---------- starting workflow 2 - Removal of new mons --------")
+    # Setting the mon service as unmanaged
+    if not mon_obj.set_mon_service_managed_type(unmanaged=True):
+        log.error("Could not set the mon service to unmanaged")
+        raise Exception("mon service not unmanaged error")
+
+    # Removing mon service
+    if not mon_obj.remove_mon_service(host=alt_mon_host.hostname):
+        log.error("Could not remove the new mon added previously")
+        raise Exception("mon service not removed error")
+
+    # Setting the mon service as managed by cephadm
+    if not mon_obj.set_mon_service_managed_type(unmanaged=False):
+        log.error("Could not set the mon service to managed")
+        raise Exception("mon service not managed error")
+
+    quorum = mon_obj.get_mon_quorum_hosts()
+    if alt_mon_host.hostname in quorum:
+        log.error(
+            f"selected host : {alt_mon_host.hostname} is present as part of Quorum post removal"
+        )
+        raise Exception("Mon in quorum error post removal")
+
+    log.info("Completed Scenario 2")
+
+    log.info(
+        "---------- starting workflow 3 - Removal of mons during OSD down --------"
```

Comment:
Why would this workflow or behaviour be any different from the second one?

Reply:
Here the removal is during OSD down, which would have a different cluster map and mon map.

The goal here is to check if the new cluster maps and mon maps get generated with mons and OSDs are removed, and cluster would still maintain it's sanity.